packages/browser-wallet/src/popup/store/utils.ts

```diff
@@ -1,26 +1,42 @@
+import {
+    StorageAccessor,
+    storedCredentials,
+    storedJsonRpcUrl,
+    storedSelectedAccount,
+    storedUrlWhitelist,
+} from '@shared/storage/access';
+import { ChromeStorageKey } from '@shared/storage/types';
 import { atom } from 'jotai';
 
-export enum ChromeStorageKey {
-    Credentials = 'credentials',
-    JsonRpcUrl = 'jsonRpcUrl',
-}
+const accessorMap = {
+    [ChromeStorageKey.Credentials]: storedCredentials,
+    [ChromeStorageKey.SelectedAccount]: storedSelectedAccount,
+    [ChromeStorageKey.JsonRpcUrl]: storedJsonRpcUrl,
+    [ChromeStorageKey.UrlWhitelist]: storedUrlWhitelist,
+};
 
 /**
  * @description
  * Create an atom that automatically syncs with chrome local storage.
  */
 export const atomWithChromeStorage = <V>(key: ChromeStorageKey, fallback: V) => {
+    const accessor = accessorMap[key] as unknown as StorageAccessor<V>;
```

Comment:
Well, if it's not there you get a union of the different "accessors", which is obviously not accepted by the compiler. I don't know how to avoid the cast. Do you have any ideas? 😄